src/AppInstallerCLICore/Workflows/MSStoreInstallerHandler.cpp

```diff
@@ -70,37 +70,56 @@ namespace AppInstaller::CLI::Workflow
             return errorCode;
         }
 
-        bool GetFreeUserEntitlement(Execution::Context& context, const std::wstring& productId)
+        bool GetFreeEntitlement(Execution::Context& context, const std::wstring& productId)
         {
             AppInstallManager installManager;
 
             // Verifying/Acquiring product ownership
             context.Reporter.Info() << Resource::String::MSStoreInstallTryGetEntitlement << std::endl;
-            GetEntitlementResult enr = installManager.GetFreeUserEntitlementAsync(productId, winrt::hstring(), winrt::hstring()).get();
-
-            if (enr.Status() == GetEntitlementStatus::Succeeded)
+            GetEntitlementResult userEntitlementResult = installManager.GetFreeUserEntitlementAsync(productId, winrt::hstring(), winrt::hstring()).get();
+            if (userEntitlementResult.Status() == GetEntitlementStatus::Succeeded)
             {
                 context.Reporter.Info() << Resource::String::MSStoreInstallGetEntitlementSuccess << std::endl;
-                AICLI_LOG(CLI, Error, << "Get entitlement succeeded.");
-
+                AICLI_LOG(CLI, Error, << "Get user entitlement succeeded.");
             }
-            else if (enr.Status() == GetEntitlementStatus::NoStoreAccount)
+            else if (userEntitlementResult.Status() == GetEntitlementStatus::NoStoreAccount)
             {
                 context.Reporter.Info() << Resource::String::MSStoreInstallGetEntitlementNoStoreAccount << std::endl;
-                AICLI_LOG(CLI, Error, << "Get entitlement failed. No Store account.");
+                // Fallback to Device Entitlement
+                GetEntitlementResult deviceEntitlementResult = installManager.GetFreeDeviceEntitlementAsync(productId, winrt::hstring(), winrt::hstring()).get();
+                if (deviceEntitlementResult.Status() == GetEntitlementStatus::Succeeded)
+                {
+                    context.Reporter.Info() << Resource::String::MSStoreInstallGetEntitlementSuccess << std::endl;
+                    AICLI_LOG(CLI, Error, << "Get device entitlement succeeded.");
+                }
+                else if (deviceEntitlementResult.Status() == GetEntitlementStatus::NoStoreAccount)

```

Comment:
Here I am not sure if `GetFreeDeviceEntitlementAsync` can return a result with `deviceEntitlementResult.Status() == GetEntitlementStatus::NoStoreAccount`.

If this is not possible, I will remove this check